### ISSUE TITLE
Update on DPM Version and/or an iFrames Version

### ISSUE THREAD

**Issue by shaigluskin** [iATSPayments/com.iatspayments.civicrm#179]
Hi Alan,

[I asked back in June](https://github.com/iATSPayments/com.iatspayments.civicrm/issues/140) about a DPM version of iATS for CiviCRM. I'm checking in about progress for that.

I'm also curious if there is anything in the works to create a version that would satisfy SAQ-A eligibility (Stripe provides that).

Thanks,

Shai

**Comment by adixon** [comment]
Yeah, good point. The 4.7 compatibility stuff had us put the DPM code (which you can see [here](https://github.com/iATSPayments/com.iatspayments.civicrm/tree/dpm-as-transfer-checkout) by the way) on hold.

I'm curious about your reference to SAQ eligibility. From what I understand, unless the checkout page is entirely hosted by the payment processor (e.g. paypal standard), you've got at least SAQ-A-EP, which is what I assume stripe would require. The primary goal of the DPM version is definitely about updating the SAQ requirement to A-EP. Here's a useful link: (https://www.pcicomplianceguide.org/saq-a-vs-a-ep-what-e-commerce-merchants-service-providers-need-to-know-now/)

**Comment by shaigluskin** [comment]
Hi Alan,

[This article from June, 2016](https://blog.sucuri.net/2016/06/navigating-pci-self-assessment-questionnaires-saq-ecommerce-websites.html) says that "Stripe Checkout" and "Stripe.js" are iFrame solutions that qualify for SAQ-A.

Shai

**Comment by adixon** [comment]
Fascinating, thanks for that. One thing that sets stripe apart is that they host the js that is responsible for processing the cc and the redirect, so maybe that's why they can get away with SAQ-A. So the short answer to your original question is no.

**Comment by zkrebs** [comment]
Is there any update to this? We are currently needing to become PCI Compliant. IATS set us up with www.pcismart.com to determine or SAQ level. It seems like we might be able to use SAQ-A-EP, but I am not sure how storing the CC token or last 4 digits of CC or expiration date might influence that choice, it seems to say "no card holder data".

**Comment by adixon** [comment]
At this point, the versions before 1.7 are only going to qualify you for SAQ-D (like most other on-site payment processing plugins). We do have a new version in testing that will get you SAQ-A (i.e. like stripe, because the credentials get input into an iframe hosted with iATS).

But none of this is related to cc token/last 4 digits/expiration date. Those pieces of data can be handled by your CiviCRM server and still allow you to qualify for SAQ-A.

**Comment by zkrebs** [comment]
Awesome. We also use the DPM feature of the Drupal Commerce iATS module so I think that qualifies for SAQ-A as well. Any timeframe? I am going to go ahead with the SAQ-C or D for now.

**Comment by adixon** [comment]
DPM does not give you SAQ-A, only at most SAQ-A-EP, because the full credentials are entered on a form managed by your site (even though DPM posts offsite).

re: timeframe - I'm hoping in the next few weeks.

**Comment by zkrebs** [comment]
Thanks for that clarification! And thanks for the work. Been filling out the SAQ-C and yes...it's...something special :)
